### PR TITLE
refactor: bunify `grpcutil.GetUser`.

### PR DESCRIPTION
**Pull request by ioga**
## Description

Migrate `grpcutil.GetUser` method to bun to make it compatible with new-style `APIServer`s.

## Commentary (optional)

- Deprecated, duplicated methods in `internal/db/postgres_users.go` can be refactored away - will do when my macbook stops thermal throttling.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

**Review by maxrussell — APPROVED** [review]
LGTM

File: master/internal/grpcutil/auth.go [hunk 1]
```diff
@@ -41,8 +46,30 @@ var (
...
+
+	var session model.AllocationSession
+	err := v2.Verify(token, db.GetTokenKeys().PublicKey, &session, nil)
+	if err != nil {
+		log.WithError(err).Debug("failed to verify allocation_session token")
+		return nil, db.ErrNotFound
+	}
+
+	err = db.Bun().NewSelect().Model(&session).Where("id = ?", session.ID).Scan(context.Background())
+	if errors.Cause(err) == db.ErrNotFound {
```
Comment (NicholasBlaskey) [inline master/internal/grpcutil/auth.go:60]: bun will return something like ```sql.ErrNoRows``` so this ```db.ErrNotFound``` will never go off though in this case it is super unlikely to happen but something to keep in mind as we move towards bun.

**Review by NicholasBlaskey — APPROVED** [review]
nice

File: master/pkg/model/task_session.go [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 
+import "github.com/uptrace/bun"
+
 // AllocationSession corresponds to a row in the "allocation_sessions" DB table.
 type AllocationSession struct {
-	ID           SessionID    `db:"id" json:"id"`
-	AllocationID AllocationID `db:"allocation_id" json:"allocation_id"`
-	OwnerID      *UserID      `db:"owner_id" json:"owner_id"`
+	bun.BaseModel `bun:"table:allocation_sessions"`
+	ID            SessionID    `db:"id" json:"id"`
```
[the remark this answers is not in the document]
Comment (ioga) [reply master/pkg/model/task_session.go:8]: it infers it from the field name, and it seems to be matching in this struct.

File: master/internal/user/postgres_users.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/guregu/null.v3"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// UserByID returns the full user for a given ID.
+func UserByID(userID model.UserID) (*model.FullUser, error) {
```
[the remark this answers is not in the document]
Comment (ioga) [reply master/internal/user/postgres_users.go:21]: we haven't been handling this before. ideally yes, we should add it. not sure if it's a must within the scope of this PR. Maybe do it together with refactoring away the rest of `internal/db/postgres_users.go`?

File: master/internal/grpcutil/auth.go (hunk 1 above)
Comment (ioga) [reply master/internal/grpcutil/auth.go:60]: good catch

File: master/internal/user/postgres_users.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (NicholasBlaskey) [reply master/internal/user/postgres_users.go:21]: Leaving it for future refactoring makes sense to me.

**Pull request merged**